src/radical/pilot/pilot_manager.py

```diff
@@ -426,40 +419,9 @@ def _pilot_staging_input(self, sds):
         sds  = expand_staging_directives(sds)
         uids = [sd['uid'] for sd in sds]
 
-        # prepare to wait for completion
-        with self._sds_lock:
-
-            for sd in sds:
-                sd['state'] = rps.NEW
-                self._active_sds[sd['uid']] = sd
-
-            sd_states = [sd['state'] for sd
-                                     in  self._active_sds.values()
-                                     if  sd['uid'] in uids]
-        # push them out
-        self._stager_queue.put(sds)
-
-        while rps.NEW in sd_states:
-            time.sleep(1.0)
-            with self._sds_lock:
-                sd_states = [sd['state'] for sd
-                                         in  self._active_sds.values()
-                                         if  sd['uid'] in uids]
-        try:
-            if rps.FAILED in sd_states:
-                errs = list()
-                for uid in uids:
-                    if self._active_sds[uid].get('exception'):
-                        errs.append(self._active_sds[uid]['exception'])
-
-                if errs:
-                    raise RuntimeError('pilot staging failed: %s' % errs)
-                else:
-                    raise RuntimeError('pilot staging failed')
-        finally:
-            with self._sds_lock:
-                for uid in uids:
-                    del self._active_sds[uid]
+        for sd in sds:
+            sd['state'] = rps.NEW
```

Comment:
seems like we don't need to keep track of the state for `sds`, since all these operations will be executed here (and not being sent to some other component)

Reply:
Yep.